### PR TITLE
Improve CommandInvocationIntrinsics API documentation and style

### PR DESCRIPTION
**Pull request by rjmholt**
# PR Summary

Using this API, I found I had to inspect the source code to determine the behaviour of the overloads I was looking over.

I've added comments so that in future that won't be necessary.

**I think it's particularly interesting that the `string` overloads execute in a local scope, while the `ScriptBlock` overloads dot-source &mdash; so this seems like an important thing to capture in documentation.**

I've also hung the parameters ~and renamed one parameter because it was inconsistently named across overloads~ (reverted this second part since it's a breaking change for named parameters).

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/engine/MshCmdlet.cs [hunk 1]
```diff
@@ -679,31 +680,36 @@ internal IEnumerable<CommandInfo> GetCommands(string name, CommandTypes commandT
         /// <exception cref="FlowControlException"></exception>
         public Collection<PSObject> InvokeScript(string script)
         {
-            return InvokeScript(script, true, PipelineResultTypes.None, null);
+            return InvokeScript(script, useNewScope: true, PipelineResultTypes.None, null);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/MshCmdlet.cs:683]: Could you please add a name for null argument too?

File: src/System.Management.Automation/engine/MshCmdlet.cs [hunk 2]
```diff
@@ -679,31 +680,36 @@ internal IEnumerable<CommandInfo> GetCommands(string name, CommandTypes commandT
...
-        /// <param name="args">The arguments to the script.</param>
+        /// <param name="args">The arguments to the script, available as $args.</param>
         /// <returns>A collection of MshCobjects generated by the script.</returns>
         /// <exception cref="ParseException">Thrown if there was a parsing error in the script.</exception>
         /// <exception cref="RuntimeException">Represents a script-level exception.</exception>
         /// <exception cref="FlowControlException"></exception>
         public Collection<PSObject> InvokeScript(string script, params object[] args)
         {
-            return InvokeScript(script, true, PipelineResultTypes.None, null, args);
+            return InvokeScript(script, useNewScope: true, PipelineResultTypes.None, null, args);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/MshCmdlet.cs:698]: The same.

File: src/System.Management.Automation/engine/MshCmdlet.cs [hunk 3]
```diff
@@ -679,31 +680,36 @@ internal IEnumerable<CommandInfo> GetCommands(string name, CommandTypes commandT
...
+        /// The scriptblock will be executed in the calling scope (dot-sourced) rather than in a new child scope.
         /// </summary>
-        /// <param name="sessionState"></param>
-        /// <param name="scriptBlock"></param>
-        /// <param name="args"></param>
-        /// <returns></returns>
+        /// <param name="sessionState">The session state in which to execute the scriptblock.</param>
+        /// <param name="scriptBlock">The scriptblock to execute.</param>
+        /// <param name="args">The arguments to the scriptblock, available as $args.</param>
+        /// <returns>A collection of the PSObjects emitted by the executing scriptblock.</returns>
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/MshCmdlet.cs:708]: Can be null or no? It would help to nullable annotations in future.

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@rjmholt @TravisEz13 Any changes needed for this PR?

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Pull request merged**